### ISSUE TITLE
Implement a check to verify that all QR codes were found (none are skipped)

### ISSUE THREAD

**Issue by jose1711** [intra2net/paperbackup#11]
Even with a perfect QR code it could happen that `zbarimg` fails to detect it (https://github.com/mchehab/zbar/issues/65). Since the codes are numbered it would be nice if `paperrestore.sh` contains a check whether there is no gap in the sequence.

**Comment by intra2net** [comment]
Good idea.

We tried to keep the restore code very simplistic so that it can easily be ported to other systems (think restoring an old backup several years in the future).

I think to implement it we'd have to break the pipe in paperrestore after the sort, store the sorted results of zbarimg in a tempfile and then check the contents (compare wc -l vs. the barcode number in the last line). 

Or do you see a way to do it without a tempfile?

**Comment by jose1711** [comment]
One idea would be to use `awk`. With `NR` special variable it could do a check for each line - whether it matches the number at the beginning of each line. The next question would be whether the error is just reported and processing terminated, data is skipped or empty values are filled with zeroes.

**Comment by jose1711** [comment]
Here a POC:
```
/usr/bin/zbarimg --raw -Sdisable -Sqrcode.enable "$SCANNEDFILE" 2>/dev/null \
    | sed -e "s/\^/\x0/g" \
    | sort -z -n \
    | awk 'BEGIN{ORS="\0"; RS="\0"} $1 {lines[$1]=$0; last=$1} END{printf "%s", "\0"; for(i=1; i<=last;i++){if (lines[i]){print lines[i]} else {print i" MISSING_DATA\n"}}}' \
    | sed ':a;N;$!ba;s/\n\x0[0-9]* //g;s/\x0[0-9]* //g;s/\n\x0//g'
```

**Comment by anarcat** [comment]
i agree with @intra2net that the restore code should be simple, or at least that there should be a simple version of it, ideally printed alongside the document (#16).

that said, if we're going to add awk to the already complex and difficult to read pipeline here, i think we'd be better serve by adding a restore option to the python program itself, keeping a fallback copy as a shell script.
